### PR TITLE
[bugfix] Fix timing in the allocation speed test

### LINKED ISSUE #802: Alloc speed check reported timings seem problematic
Using Clang 7 and `-O0` I get negative timings:

```
==============================================================================
PERFORMANCE REPORT
------------------------------------------------------------------------------
AllocSpeedTest_no
- tresa:login
   - PrgEnv-clang
      * time: 0.441987 s
------------------------------------------------------------------------------
AllocSpeedTest_2M
   - PrgEnv-clang
      * time: -0.332334 s
------------------------------------------------------------------------------
```

Using `-O3`, I get zero seconds.

```
clang version 7.0.1 (tags/RELEASE_701/final)
Target: x86_64-apple-darwin18.5.0
Thread model: posix
InstalledDir: /usr/local/opt/llvm/bin
```

### PR DESCRIPTION
**Pull request by teojgo**
Fixes #802

### CODE REVIEW

**Comment by teojgo** [comment]
@vkarak can you check that this fixes the problem in the timings for your case ?

**Comment by toxa81** [comment]
looks ok

**Comment by teojgo** [comment]
@jenkins-cscs retry all

File: cscs-checks/microbenchmarks/alloc_speed/src/alloc_speed.cpp [hunk 1]
```diff
@@ -1,21 +1,25 @@
...
-#include <memory>
 
 double test_alloc(size_t n)
 {
     auto t_start = std::chrono::system_clock::now();
-    std::unique_ptr<char[]> ptr(new char[n]);
 
     /* time to allocate + fill */
-    std::fill(ptr.get(), ptr.get() + n, 0);
+    char* ptr = (char*)std::malloc(n);
```
Comment (vkarak) [inline cscs-checks/microbenchmarks/alloc_speed/src/alloc_speed.cpp]: I would add a small comment with the rationale of not using `unique_ptr` here.

**Comment by vkarak** [comment]
@jenkins-cscs retry kesch

**Pull request merged**